src/test/scala/io/gatling/build/versioning/GatlingBumpSpec.scala

```diff
@@ -0,0 +1,66 @@
+    (s"$YYYY.$WW.0"               , s"$YYYY.$WW.1"      , s"$YYYY.${WW + 1}.0"      , s"$YYYY.$WW.1", s"$YYYY.$WW.1-M$YYYYMMDDHHMMSS"      , s"$YYYY.${WW + 1}.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.1-M$YYYYMMDDHHMMSS"),
+    // milestones
+    ("1.0.0-M42"                  , "1.0.0"             , "1.0.0"                   , s"$YYYY.$WW.0", s"1.0.0-M$YYYYMMDDHHMMSS"            , s"1.0.0-M$YYYYMMDDHHMMSS"                  , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    ("1.0.1-M42"                  , "1.0.1"             , "1.1.0"                   , s"$YYYY.$WW.0", s"1.0.1-M$YYYYMMDDHHMMSS"            , s"1.1.0-M$YYYYMMDDHHMMSS"                  , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    ("1.2.3-M42"                  , "1.2.3"             , "1.3.0"                   , s"$YYYY.$WW.0", s"1.2.3-M$YYYYMMDDHHMMSS"            , s"1.3.0-M$YYYYMMDDHHMMSS"                  , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    (s"${YYYY - 1}.$WW.0-M42"     , s"${YYYY - 1}.$WW.0", s"${YYYY - 1}.$WW.0"      , s"$YYYY.$WW.0", s"${YYYY - 1}.$WW.0-M$YYYYMMDDHHMMSS", s"${YYYY - 1}.$WW.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    (s"$YYYY.$WW.0-M42"           , s"$YYYY.$WW.0"      , s"$YYYY.$WW.0"            , s"$YYYY.$WW.0", s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"            , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    (s"$YYYY.$WW.1-M42"           , s"$YYYY.$WW.1"      , s"$YYYY.${WW + 1}.0"      , s"$YYYY.$WW.1", s"$YYYY.$WW.1-M$YYYYMMDDHHMMSS"      , s"$YYYY.${WW + 1}.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.1-M$YYYYMMDDHHMMSS"),
+    // snapshots same rules as milestones
+    ("1.0.0-SNAPSHOT"             , "1.0.0"             , "1.0.0"                   , s"$YYYY.$WW.0", s"1.0.0-M$YYYYMMDDHHMMSS"            , s"1.0.0-M$YYYYMMDDHHMMSS"                  , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    ("1.0.1-SNAPSHOT"             , "1.0.1"             , "1.1.0"                   , s"$YYYY.$WW.0", s"1.0.1-M$YYYYMMDDHHMMSS"            , s"1.1.0-M$YYYYMMDDHHMMSS"                  , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    ("1.2.3-SNAPSHOT"             , "1.2.3"             , "1.3.0"                   , s"$YYYY.$WW.0", s"1.2.3-M$YYYYMMDDHHMMSS"            , s"1.3.0-M$YYYYMMDDHHMMSS"                  , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    (s"${YYYY - 1}.$WW.0-SNAPSHOT", s"${YYYY - 1}.$WW.0", s"${YYYY - 1}.$WW.0"      , s"$YYYY.$WW.0", s"${YYYY - 1}.$WW.0-M$YYYYMMDDHHMMSS", s"${YYYY - 1}.$WW.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    (s"$YYYY.$WW.0-SNAPSHOT"      , s"$YYYY.$WW.0"      , s"$YYYY.$WW.0"            , s"$YYYY.$WW.0", s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"            , s"$YYYY.$WW.0-M$YYYYMMDDHHMMSS"),
+    (s"$YYYY.$WW.1-SNAPSHOT"      , s"$YYYY.$WW.1"      , s"$YYYY.${WW + 1}.0"      , s"$YYYY.$WW.1", s"$YYYY.$WW.1-M$YYYYMMDDHHMMSS"      , s"$YYYY.${WW + 1}.0-M$YYYYMMDDHHMMSS"      , s"$YYYY.$WW.1-M$YYYYMMDDHHMMSS"),
+    // format: ON
+  )
```

Comment:
@slandelle Please ensure that I correctly understand what we should have.
If this table is not readable enough, we can live review it together.

Reply:
If I get it right, from now on, milestones with have a date instead of an incremental number, right?
Are you missing the specs where the old version is patch-milestone or minor-milestone? Or are those cases not possible because they are not considered/filtered out because the tag is a lightweight one?

Reply:
Report of our discussion:

**Milestone number**
This plugin had always generated milestone number based on current date (hard to reliably find the latest milestone based on a version inside different branch).
_Note: precision is second_

**Manual steps**
If needed, we still can create lightweight tag (for milestone, rc...) or annotated tag (for stable) manually.

**Lightweight vs annotated**
* We can have multiple tags on the current HEAD (recreating another milestone in the same commit because of a flaky failure in CI, for instance)
* Computation from a Milestone qualified version should be the norm: ie, we should deploy a milestone in dev. If successful, we then create a stable release to deploy in preprod, then in prod
* Lightweight tags can be filtered out. And they should be deleted after a small time unused.
* Annotated tags should stay forever